docs/content/docs/connectors/datastream/pulsar.md

```diff
@@ -387,27 +391,402 @@ You should enable transaction in the Pulsar `borker.conf` file when using these
+        <td>Gauge</td>
+    </tr>
+    <tr>
+        <td>PulsarSink.numAcksReceived</td>
+        <td>n/a</td>
+        <td>The number of acks received for sent messages.</td>
+        <td>Counter</td>
+    </tr>
+    <tr>
+        <td>PulsarSink.sendLatencyMax</td>
+        <td>n/a</td>
+        <td>The maximum send latency in the last refresh interval across all producers.</td>
+        <td>Gauge</td>
+    </tr>
+    <tr>
+        <td>PulsarSink.producer."ProducerName".sendLatency50Pct</td>
+        <td>ProducerName</td>
+        <td>The 50th percentile of send latency in the last refresh interval for a specific producer.</td>
+        <td>Gauge</td>
+    </tr>
+    <tr>
+        <td>PulsarSink.producer."ProducerName".sendLatency75Pct</td>
+        <td>ProducerName</td>
+        <td>The 75th percentile of send latency in the last refresh interval for a specific producer.</td>
+        <td>Gauge</td>
+    </tr>
+    <tr>
+        <td>PulsarSink.producer."ProducerName".sendLatency95Pct</td>
+        <td>ProducerName</td>
+        <td>The 95th percentile of send latency in the last refresh interval for a specific producer.</td>
+        <td>Gauge</td>
+    </tr>
+    <tr>
+        <td>PulsarSink.producer."ProducerName".sendLatency99Pct</td>
+        <td>ProducerName</td>
+        <td>The 99th percentile of send latency in the last refresh interval for a specific producer.</td>
+        <td>Gauge</td>
+    </tr>
+    <tr>
+        <td>PulsarSink.producer."ProducerName".sendLatency999Pct</td>
+        <td>ProducerName</td>
+        <td>The 99.9th percentile of send latency in the last refresh interval for a specific producer.</td>
+        <td>Gauge</td>
+    </tr>
+  </tbody>
+</table>
+
+{{< hint info >}}
+- `numBytesOut`, `numRecordsOut`, `numRecordsOutErrors` are retrieved from Pulsar client metrics.
+
+- `currentSendTime` tracks the time from when the producer calls `sendAync()` to
+  the time when the message is acknowledged by the broker. This metric is not available in `NONE` delivery guarantee.
+{{< /hint >}}
+
+The Pulsar producer refreshes its stats every 60 seconds by default. And PulsarSink retrieves the Pulsar producer
+stats every 500ms. Thus `numRecordsOut`, `numBytesOut`, `numAcksReceived`, and `numRecordsOutErrors` 
+is updated every 60 seconds. To increase the metrics refresh frequency, you can change
+the Pulsar producer stats refresh interval to a smaller value (minimum 1 second), as shown below.
+
+```java
+builder.setConfig(PulsarOptions.PULSAR_STATS_INTERVAL_SECONDS. 1L)
+```
+
+`numBytesOutRate` and `numRecordsOutRate` are calculated based on the `numBytesOut` and `numRecordsOUt`
+counter respectively. Flink internally uses a fixed 60 seconds window to calculate the rates.
+
+### Brief Design Rationale
+
+Pulsar sink follow the Sink API defined in 
+[FLIP-191](https://cwiki.apache.org/confluence/display/FLINK/FLIP-191%3A+Extend+unified+Sink+interface+to+support+small+file+compaction).
```

Comment:
I think you mean FLIP-143? https://cwiki.apache.org/confluence/display/FLINK/FLIP-143%3A+Unified+Sink+API

Reply:
Well the PulsarSInk is based on SinkV2 which is the FLIP-191 mentioned API